data/pigui/views/game.lua

```diff
@@ -285,19 +287,21 @@ ui.registerHandler('game', function(delta_t)
 		-- TODO: dispatch escape key to views and let them handle it
 		if currentView == "world" and ui.escapeKeyReleased(true) then
 			if not ui.optionsWindow.isOpen then
+			    lastTimeAcceleration = Game.GetTimeAcceleration() ~= Game.GetRequestedTimeAcceleration() and Game.GetRequestedTimeAcceleration() or Game.GetTimeAcceleration()
 				Game.SetTimeAcceleration("paused")
 				ui.optionsWindow:open()
 				Input.EnableBindings(false)
 			else
 				ui.optionsWindow:close()
-				if not ui.optionsWindow.isOpen then
-					Game.SetTimeAcceleration("1x")
-					Input.EnableBindings()
-				end
 			end
 		end
 
 		callModules('modal')
+		
+		if not ui.optionsWindow.isOpen then
+			Game.SetTimeAcceleration((lastTimeAcceleration == "paused") and "1x" or lastTimeAcceleration)
+			Input.EnableBindings()
+		end
```

Comment:
This block will be called every frame that the options window is closed. The code should either perform a boundary check (was-open to not-open) or the settings window should queue an `onPauseMenuOpen`/`onPauseMenuClosed` event of some sort and this logic be moved inside that event handler.

Reply:
Oof. Now I remember why I added `settingsMenuWasOpened` variable. But I like you solution with event more. I'll see what I can do.

Reply:
On the fence here. While game is paused events are not proceeded. So I can do `Game.SetTimeAcceleration("1x")` before or after `Event.Queue("onPauseMenuClosed")` and this function will be completed but you will see game setting speed to `1x` for brief second. Or I can do `Event._Emit()` but it says in `Event.lua` that it's an internal method that is being called from C++ so I suppose it's not supposed to be called from lua. What do you think is the best approach?

Reply:
Realistically, we need two event queues - one for "world" events that only process while the simulation is being ticked, and one for "game" events like playing music and reacting to the UI. I'd recommend the first solution you proposed; 1x time acceleration for a few frames is not a big deal (and I'd like to smoothly accelerate to the newly-requested time accel value anyways), and it can take advantage of future changes to the Event architecture.

Reply:
Force-pushed with my first approach with `Game.SetTimeAcceleration("1x") `

Reply:
Clicking the "close" button in the Pause menu doesn't resume time acceleration and leaves the game paused. Pressing ESC to close the pause menu does. Otherwise, looks good.

Reply:
Had to include `Event` in `settings-window.lua` and added this bit to `Close` button in the same file:
```
if Game.player then
	Game.SetTimeAcceleration("1x")
	 Event.Queue("onPauseMenuClosed")
end
```
Let this be a temporary solution before two event queues will be implemented in the near future.